test/Microsoft.ML.Functional.Tests/Common.cs

```diff
@@ -219,6 +227,38 @@ public static void AssertMetrics(MulticlassClassificationMetrics metrics)
             Assert.InRange(metrics.MicroAccuracy, 0, 1);
             Assert.True(metrics.LogLoss >= 0);
             Assert.InRange(metrics.TopKAccuracy, 0, 1);
+
+            // Confusion matrix validations
+            Assert.NotNull(metrics.ConfusionMatrix);
+            AssertConfusionMatrix(metrics.ConfusionMatrix);
+           
+        }
+
+        internal static void AssertConfusionMatrix(ConfusionMatrix confusionMatrix)
+        {
+            // Confusion matrix validations
+            Assert.NotNull(confusionMatrix);
+            Assert.NotEmpty(confusionMatrix.Counts);
+            Assert.NotEmpty(confusionMatrix.PerClassPrecision);
+            Assert.NotEmpty(confusionMatrix.PerClassRecall);
+            Assert.NotNull(confusionMatrix.ClassIndicators);
+
+            foreach (var precision in confusionMatrix.PerClassPrecision)
+                Assert.InRange(precision, 0, 1);
+
+            foreach (var recall in confusionMatrix.PerClassRecall)
+                Assert.InRange(recall, 0, 1);
+
+
+            // Get the values in the annotations
```

Comment:
DOUBLE LINES! DOUBLE SHOCK!

Reply:
ofc it'll happen, there is not compiler complaining about it :)

---
In reply to: [275464491](https://github.com/dotnet/machinelearning/pull/3250#discussion_r275464491) [](ancestors = 275464491)